gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileStatus.java

```diff
@@ -0,0 +1,61 @@
+package com.google.cloud.hadoop.fs.gcs;
+
+import com.google.cloud.hadoop.gcsio.FileInfo;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageItemInfo;
+import com.google.cloud.hadoop.gcsio.StorageResourceId;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsPermission;
+
+public class GoogleHadoopFileStatus extends FileStatus {
+  private final GoogleCloudStorageItemInfo itemInfo;
+
+  GoogleHadoopFileStatus(
+      FileInfo fileInfo,
```

Comment:
We could rely on `itemInfo` directly out of fileInfo instead of Creating a new object

Reply:
That method is protected, I will change it to public and then call the method